### PR TITLE
Home Page Configurability for Front End

### PR DESCRIPTION
**Pull request by uglycoyote**
* [ n/a ? ] I have updated the unit tests
* [ X ] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [ no ] I have updated the appropriate documentation

Please suggest a file where the documentation change should be made, if applicable.

### CODE REVIEW

**Comment by uglycoyote** [comment]
sorry about the messy list of commits here, we had a bit of pollution in our master branch from an older change.  It's really just the last two commits which are relevant to this change.

**Comment by uglycoyote** [comment]
FYI  @sphet brought up these changes on IRC and got recommendations from @tardyp  about the best way to do it

File: master/buildbot/newsfragments/home-page-configurability.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Web front end now allows you to configure the default landing page with `c['www']['default_page'] = 'name-of-page'`.
```
Comment (tardyp) [inline master/buildbot/newsfragments/home-page-configurability.feature]: This needs to be documented in the doc (not only release notes)

**Comment by uglycoyote** [comment]
@tardyp  can you suggest where the appropriate place to document this would be?

Also, both @sphet and myself are Perforce users and don't know git very well. I spent a good four hours yesterday trying to get this commit not to include junk from @sphet's 2018 p4poller change (which was already accepted, but still somehow github wants to merge a second time), and spent a few hours today investigating your suggestion of `git rebase -i` but that led me to no end of confusion (the interactive session gave me a lots of errors and roadblocks while trying to do this).  I read of ton of git material and feel no closer to getting to a solution with this, I'm going to put it aside for now, and perhaps revisit this later.  Let me know if you have any more specific suggestion as to how to clean this up.

Thanks

**Comment by uglycoyote** [comment]
To be a bit more informative about the kinds of troubles that I run into, it's mostly merge conflicts on files that we haven't touched, I'm not equipped with the information to know the correct resolution, and I don't really understand why I need to resolve those as part of the rebase operation, when presumably other people have already resolve them in the past.

here's what I am trying:

* The commit list above contains both old and new commits 
* If I want to squash all of those, my understanding is that  I need to start the rebase at the change before the oldest one that appears above,  `e1d5fee`, which is `9f54f5390466019db795c6545280c245079dea62`
* I do `git rebase -i 9f54f5390466019db795c6545280c245079dea62` and edit the log to squash and/or remove the changes which are appearing in this PR that I don't want
* It starts rebasing 2057 commits which is everything that has happened on buildbot's master branch in the last year
* I get a conflict after a few hundred, dealing with a change from @rodrigc 

```
Auto-merging common/pylintrc
CONFLICT (content): Merge conflict in common/pylintrc
error: could not apply bf6ace4dc... pylint: reenable self-cls-assignment warning
```

In this case when I look at the conflict, it's two different changes trying to add a different line in the same spot in `pylintrc`

```
  consider-using-dict-comprehension,
  consider-using-set-comprehension,
  assignment-from-none,
<<<<<<< HEAD
  self-cls-assignment,
=======
  useless-import-alias,
>>>>>>> bf6ace4dc... pylint: reenable self-cls-assignment warning
  trailing-comma-tuple,
  comparison-with-callable,
  comparison-with-itself
```

* In this case it seems like a fairly trivial conflict and I accept both of the newly added lines, and continue on (without being certain that it was the correct way to resolve)
* I get further conflicts in `pylintrc`, which i resolve similarly
* eventually I get to hairier conflicts, e.g. in guanlecoja's readme file, which I cannot guess at a proper resolution for, which leaves me stuck and unable to continue the rebase 

Any suggestions about how I might be able to squash or delete these year-old commits without needing to involve the entire team of developers to figure out the correct course of action on these resolves?  

Thanks, and apologies for my lack of git skills here.   I have probably read about 100 different blogs/manpages/etc on the subject over the last two days, and feel no closer to a resolution here.

I certainly could have saved a lot of time if I  just set up a new github account and re-fork, and re-apply our small set of changes in a tidier way, though I would like to learn the proper way to fix this situation, so any help with this would be appreciated.

**Comment by tardyp** [comment]
Here are the list of commands you can do do restart from  a proper environment
```
git remote add buildbot https://github.com/buildbot/buildbot/ . # add buildbot as a remote for you local tree
git fetch --all . # make sure we have the last version of buildbot/master
git checkout buildbot/master -b home_page_configurability
git cherry-pick e1d5fee 3032f3b 54567ae 9e7f670 . # cherry-pick commit meaningful
git rebase -i buildbot/master  # then squash the commit together by replacing most 'pick' words with 'fixup'
```

If you have one tutorial to follow, please follow this one: https://learngitbranching.js.org/

```
git fetch --all
git checkout buildbot/master -b home_page_configurability
```
Should be the preferred way to start a new work when contributing open-source

**Comment by uglycoyote** [comment]
@tardyp thanks for the instructions and the tutorial link.  I followed the tutorial for quite a while and learned a few things.  

I'll close this since there's now another pull request open from a different branch.

**Pull request closed without merging**